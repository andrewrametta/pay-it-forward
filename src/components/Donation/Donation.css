div.donation-wrapper {
  height: 100%;
  margin: 50px 10px;
  background: white;
  box-shadow: 2px 2px 6px 0 rgba(41, 2, 80, 0.1);
  border-radius: 10px;
  padding: 10px;
}

div.donation-img {
  text-align: center;
}

img.donation-image {
  width: 100%;
  height: 100%;
  padding: 5px;
}

div.item-container-details {
  border-bottom: 1px solid #4500c9;
  margin-left: 10px;
  color: #4500c9;
}

div.user-container-details {
  padding-top: 20px;
}

@media all and (min-width: 600px) {
  div.donation-wrapper {
    margin: 30px;
    display: flex;
    flex-direction: column;
  }

  div.desktop-flex-view {
    display: flex;
    flex-direction: row;
    border-bottom: 1px solid #4500c9;
  }

  div.item-container-details {
    border-bottom: none;
    margin-top: 20px;
  }

  div.user-container-details {
    margin-left: 20px;
  }
}
