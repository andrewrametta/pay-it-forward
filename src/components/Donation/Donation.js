import React, { useContext } from "react";
import AppContext from "../../AppContext";
import AuthAPIService from "../../services/auth-api-service";
import { Image } from "cloudinary-react";

function Donation(props) {
  const id = props.match.params.donationId;
  const {
    items,
    type,
    userId,
    conversations,
    setConversations,
    setItems,
  } = useContext(AppContext);
  const donationArray = items.filter((item) => item.id === parseInt(id));
  const donationItem = donationArray.length > 0 ? donationArray[0] : null;

  const handleConversation = (e) => {
    e.preventDefault();
    AuthAPIService.postConversation({
      user_id: userId,
      user2_id: donationItem.user_id,
    })
      .then((conversation) => {
        props.history.push("/messages");
        setConversations([...conversations, conversation]);
      })
      .catch((err) => {
        console.log(err);
      });
  };
  const handleDelete = (e) => {
    e.preventDefault();
    AuthAPIService.deleteItem(id)
      .then(() => {
        props.history.push("/yourdonations");
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  // const deleteYourItem = (item_id) => {
  //   setItems({
  //     items: items.filter((item) => item.id !== item_id),
  //   });
  //   // setItems({
  //   //   items: items.filter((item) => item.id !== item_id),
  //   // });
  // };

  return (
    <div>
      {" "}
      <h2>This is a donation</h2>
      <Image
        cloudName="hq1rpt94r"
        publicId={`${donationItem.item_url}`}
        width="150"
        height="200"
        crop="fill"
      />
      <h3>{donationItem.title}</h3>
      <p>{donationItem.description}</p>
      {type === "org" && <button onClick={handleConversation}>Request</button>}
      {userId === donationItem.user_id && (
        <button onClick={handleDelete}>Delete</button>
      )}
    </div>
  );
}

export default Donation;
