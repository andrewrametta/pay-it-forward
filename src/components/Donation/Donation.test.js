import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Donation from "./Donation";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`Donation component`, () => {
  it.only("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <Donation />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
