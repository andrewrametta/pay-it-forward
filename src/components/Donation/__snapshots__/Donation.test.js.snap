// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donation component renders the complete page 1`] = `<Donation />`;
